### PR TITLE
chore: update react 18

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1169

- React 18 related API changes (ie: React.PropsWithChildren)
- Update dependencies that require React 18 (@mui, react-slick, victory, ..)
- Replace some dependencies with no React 18 support (ie: react-html-parser)
- migrate legacy makeStyle API to tss-react
- use React 17 compatible version of `web-components` wor `web-www`

**Note for reviewer:**
Might be more important for this PR to do some regression testing (feature + visual) than review all the changes.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1518--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by wgwz** [comment]
hey @flagrede i ran through testing in the deploy preview, and i could not find any issues. i ran tests for:

- buying credits from different classes (from project pages and from trade table)
- creating/cancelling sell orders
- failure to buy credits when i didn't have enough credit
- viewing my portfolio, taking some actions from my ecocredits table
- checked out pagination in tables where we have it
- confirmed that my actions had corresponding txs shown in the activity table
- sorting on projects table
- login (obviously) and logout

i was testing from desktop chrome, LGTM! thanks for working on this a getting us upgraded, this must have been a lot of work 💪

do you have any particular code changes in mind where you think some more eyes to review could help? if you point some places out i'd be glad to review more in-depth.

**Comment by flagrede** [comment]
> hey @flagrede i ran through testing in the deploy preview, and i could not find any issues. i ran tests for:
> 
> * buying credits from different classes (from project pages and from trade table)
> * creating/cancelling sell orders
> * failure to buy credits when i didn't have enough credit
> * viewing my portfolio, taking some actions from my ecocredits table
> * checked out pagination in tables where we have it
> * confirmed that my actions had corresponding txs shown in the activity table
> * sorting on projects table
> * login (obviously) and logout
> 
> i was testing from desktop chrome, LGTM! thanks for working on this a getting us upgraded, this must have been a lot of work 💪
> 
> do you have any particular code changes in mind where you think some more eyes to review could help? if you point some places out i'd be glad to review more in-depth.

@wgwz Thanks a lot for the in-depth testing, that's super helpful!
Regarding specific changes to look at, it's hard to say. Maybe checking all the `package.json` for dependencies coherency could be helpful too. Aside from that, there are a lot of boring changes in this PR.

**Comment by wgwz** [comment]
> @wgwz Thanks a lot for the in-depth testing, that's super helpful!
Regarding specific changes to look at, it's hard to say. Maybe checking all the package.json for dependencies coherency could be helpful too. Aside from that, there are a lot of boring changes in this PR.

@flagrede could you describe more what you mean by checking for cohorency? i’d be glad to spend some time on that, i’m just not sure exactly what you mean

**Comment by flagrede** [comment]
> > @wgwz Thanks a lot for the in-depth testing, that's super helpful!
> > Regarding specific changes to look at, it's hard to say. Maybe checking all the package.json for dependencies coherency could be helpful too. Aside from that, there are a lot of boring changes in this PR.
> 
> @flagrede could you describe more what you mean by checking for cohorency? i’d be glad to spend some time on that, i’m just not sure exactly what you mean

I meant just checking for react version used across packages. Should be the same React 18 version for all packages except `web-www` that still use React 17.

File: web-auth/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 import LoginForm, {
   Values,
 } from 'web-components/lib/components/form/LoginForm';
 import OnBoardingSection from 'web-components/lib/components/section/OnBoardingSection';
 import { Theme } from 'web-components/lib/theme/muiTheme';
+import { WebAuth } from 'auth0-js';
+import axios from 'axios';
+import React, { useCallback } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { useGoogleReCaptcha } from 'react-google-recaptcha-v3';
```
Comment (blushi) [inline web-auth/src/App.tsx:10]: just a nit but we usually keep the external lib imports above the web-components/ ones

**Review by wgwz — APPROVED** [review]
Pre-approving. I'm still going to go check the dependencies like you asked.

**Review by blushi — COMMENTED** [review]
still reviewing the code
in the mean time I tried some local testing and I get this when starting gatsby website (`yarn start-www`) and the website deploy preview is just blank https://deploy-preview-1518--regen-website.netlify.app/ because of the same error I think:

image

**Review by victordelval — APPROVED** [review]
Awesome work, thanks @flagrede! :100: :rocket:

**Comment by victordelval** [comment]
I have manually tested various types of use cases with txs (project creation, credit batches, purchase, sell order, cancel sell orders, put/take from basket). I have detected some small detail but they have nothing to do with this PR!

**Comment by flagrede** [comment]
> still reviewing the code in the mean time I tried some local testing and I get this when starting gatsby website (`yarn start-www`) and the website deploy preview is just blank https://deploy-preview-1518--regen-website.netlify.app/ because of the same error I think:
> 
> image

I might have messed up something with a rebase, I'm investigating on it.

**Comment by mhagel** [comment]
Do we have the deployment CI set up for the new `web-www` repo?

**Review by mhagel — APPROVED** [review]
Thank you!!!

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**